renderdoc/driver/vulkan/vk_pixelhistory.cpp

```diff
@@ -712,7 +731,21 @@ struct VulkanOcclusionAndStencilCallback : public VulkanPixelHistoryCallback
           CreateRenderPass(pipestate.renderPass, pipestate.GetFramebuffer(), pipestate.subpass);
       VkFramebuffer newFb = CreateFramebuffer(pipestate.renderPass, newRp, pipestate.subpass,
                                               pipestate.GetFramebuffer(), m_StencilImageView);
-      PipelineReplacements replacements = GetPipelineReplacements(eid, pipestate.graphics.pipeline);
+      const VulkanCreationInfo::Framebuffer &fbInfo =
+          m_pDriver->GetDebugManager()->GetFramebufferInfo(pipestate.GetFramebuffer());
+      uint32_t framebufferIndex = 0;
+      for(uint32_t i = 0; i < fbInfo.attachments.size(); i++)
+      {
+        ResourceId img =
+            m_pDriver->GetDebugManager()->GetImageViewInfo(fbInfo.attachments[i].createdView).image;
```

Comment:
I think you should use `pipestate.GetFramebufferAttachments()` here instead, so that this works with imageless framebuffers.

Reply:
Ah, makes sense, thanks!